a3fe/run/leg.py

```diff
@@ -259,11 +259,12 @@ def setup(
                     relative_simulation_cost=self.relative_simulation_cost,
                     ensemble_size=self.ensemble_size,
                     lambda_values=cfg.lambda_values[self.leg_type][stage_type],
-                    base_dir=self.stage_input_dirs[stage_type].replace("/input", ""),
+                    base_dir=self.stage_input_dirs[stage_type].rsplit("/input", 1)[0],
                     input_dir=self.stage_input_dirs[stage_type],
-                    output_dir=self.stage_input_dirs[stage_type].replace(
-                        "input", "output"
-                    ),
+                    output_dir="/".join(
+                        self.stage_input_dirs[stage_type].split("/")[:-1]
+                    )
+                    + "/output",
```

Comment:
For clarity, here I would do:

output_dir = os.path.join(self.stage_input_dirs[stage_type], "output")